hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/snapshot/filter/TestAbstractReclaimableFilter.java

```diff
@@ -0,0 +1,325 @@
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.hdds.utils.TransactionInfo;
+import org.apache.hadoop.hdds.utils.db.DBStore;
+import org.apache.hadoop.hdds.utils.db.RDBStore;
+import org.apache.hadoop.hdds.utils.db.Table;
+import org.apache.hadoop.hdds.utils.db.managed.ManagedRocksDB;
+import org.apache.hadoop.ozone.om.KeyManager;
+import org.apache.hadoop.ozone.om.OMMetadataManager;
+import org.apache.hadoop.ozone.om.OmSnapshot;
+import org.apache.hadoop.ozone.om.OmSnapshotManager;
+import org.apache.hadoop.ozone.om.OzoneManager;
+import org.apache.hadoop.ozone.om.SnapshotChainManager;
+import org.apache.hadoop.ozone.om.helpers.BucketLayout;
+import org.apache.hadoop.ozone.om.helpers.OmBucketInfo;
+import org.apache.hadoop.ozone.om.helpers.SnapshotInfo;
+import org.apache.hadoop.ozone.om.lock.IOzoneManagerLock;
+import org.apache.hadoop.ozone.om.lock.OMLockDetails;
+import org.apache.hadoop.ozone.om.lock.OzoneManagerLock;
+import org.apache.hadoop.ozone.om.snapshot.ReferenceCounted;
+import org.apache.hadoop.ozone.om.snapshot.SnapshotCache;
+import org.apache.hadoop.ozone.om.snapshot.SnapshotDiffManager;
+import org.apache.hadoop.ozone.om.snapshot.SnapshotUtils;
+import org.apache.ozone.rocksdiff.RocksDBCheckpointDiffer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.MockedConstruction;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.rocksdb.ColumnFamilyDescriptor;
+import org.rocksdb.ColumnFamilyHandle;
+import org.rocksdb.DBOptions;
+import org.rocksdb.ReadOptions;
+import org.rocksdb.RocksDB;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.RocksIterator;
+
+/**
+ * Test class for ReclaimableFilter.
+ */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public abstract class TestAbstractReclaimableFilter {
+
+  private ReclaimableFilter reclaimableFilter;
+  private OzoneManager ozoneManager;
+  private OmSnapshotManager omSnapshotManager;
+  private AtomicReference<List<UUID>> lockIds = new AtomicReference<>(Collections.emptyList());
+  private List<String> volumes;
+  private List<String> buckets;
+  private MockedStatic<SnapshotUtils> mockedSnapshotUtils;
+  private Map<String, List<SnapshotInfo>> snapshotInfos;
+  @TempDir
+  private Path testDir;
+  private SnapshotChainManager snapshotChainManager;
+  private KeyManager keyManager;
+
+  protected abstract ReclaimableFilter initializeFilter(OzoneManager om, OmSnapshotManager snapshotManager,
+                                                        SnapshotChainManager chainManager,
+                                                        SnapshotInfo currentSnapshotInfo, KeyManager km,
+                                                        IOzoneManagerLock lock, int numberOfPreviousSnapshotsFromChain);
```

Comment:
nit: Please do not format method signature like this. Whenever visibility / return type / method name / other modifiers are changed, we would have to reindent all parameters.

Also elsewhere in the patch.

Reply:
done